lib/models/app_error.dart

```diff
@@ -0,0 +1,63 @@
+import 'dart:developer';
+
+import 'package:dio/dio.dart';
+
+class AppError {
+  final String message;
+  final ErrorType type;
+  final int? statusCode;
+  final int? code;
+
+  AppError(this.message, this.type, {this.statusCode, this.code});
+
+  @override
+  String toString() =>
+      'AppError(message: $message, type: $type, statusCode: $statusCode, code: $code)';
+
+  /// This consructor is used when the error comes from server (bad response).
+  ///
+  /// Here you can update it to handle server errors, according to the error model
+  /// comes from back end.
+  factory AppError.serverErrorParse(DioException error) {
+    log(error.response?.data?.toString() ?? "NULL",
+        name: "AppError.serverErrorParse");
+    final serverMessage = error.response?.data['message'] ?? "Not Defined";
+    final code = error.response?.data['code'];
+    return AppError("Server error: $serverMessage", ErrorType.server,
+        statusCode: error.response?.statusCode, code: code);
+  }
+}
+
+enum ErrorType {
+  network,
+  server,
+  validation,
+  format,
+  unknown,
+}
+
+class ErrorHandler {
+  static AppError handleError(Object? error) {
+    log(error.toString());
+    log(error.runtimeType.toString(), name: "Error type");
+    if (error.runtimeType == DioException) {
+      switch ((error as DioException).type) {
+        case DioExceptionType.connectionTimeout:
+        case DioExceptionType.sendTimeout:
+        case DioExceptionType.receiveTimeout:
+          return AppError('Connection Timeout', ErrorType.network);
```

Comment:
are we displaying this as part of a message to the user?
If yes, we would need to localise. Also, it would be nice to associate error codes to display in the error message in case we need to give support for incidences

Reply:
@sallymoc 

Yes, the messages appear to the user, and I have added the localization for them.

The error codes also display to the user (if they exist). The toString() function should provide a message describing the error (either from the server or internally), along with the status code (of the request) and the code (which comes from the backend).)

The current version of the toString() is like this:
`  

    if (statusCode != null) {
      errorMessage += ', ${l10nWrapper.l10n!.generalStatusCode}: $statusCode';
    }
    if (code != null) {
      errorMessage += ', ${l10nWrapper.l10n!.generalCode}: $code';
    }

    return errorMessage;
  `

For example, if there is a network disconnection (and hence no code or status code), it should appear like this:
![image](https://github.com/user-attachments/assets/bf91db02-c388-4ce5-97aa-36fa24d31717)

In another case, if the request is completed but the backend returns an error with the message "Not Found" and code 5, it should appear like this:
![image](https://github.com/user-attachments/assets/67d5e961-36ee-4a57-96c2-e58f9b243adc)

Please let me know if there are further improvements or suggestions for this part.